contracts/strategies/AeroStrategy.sol

```diff
@@ -0,0 +1,374 @@
+                minAmountA,
+                minAmountB,
+                address(this),
+                block.timestamp
+            );
+            uint256 lpBalance = IERC20(LP).balanceOf(address(this));
+            IVeloGauge(AERO_GAUGE).deposit(lpBalance);
+        }
+    }
+
+    function _calculateTokenAmounts(
+        uint256 excessWant
+    ) internal view returns (uint256 amountA, uint256 amountB) {
+        (uint256 desiredA, uint256 desiredB, ) = IVeloRouter(AERO_ROUTER)
+            .quoteAddLiquidity(
+                USDbC,
+                DAI,
+                true,
+                POOL_FACTORY,
+                excessWant / 2,
+                excessWant * 10 ** 12 / 2 
+            );
+        desiredB = desiredB / 10 ** 12;
+        uint256 sum = desiredB + desiredA;
+        amountA = excessWant * desiredA / sum;
+        amountB = excessWant - amountA;
+    }
+
+    function liquidatePosition(
+        uint256 _amountNeeded
+    ) internal override returns (uint256 _liquidatedAmount, uint256 _loss) {
+        uint256 _wantBal = want.balanceOf(address(this));
+        if (_wantBal >= _amountNeeded) {
+            return (_amountNeeded, 0);
+        }
+        _withdrawSome(_amountNeeded - _wantBal);
+        _wantBal = want.balanceOf(address(this));
+
+        if (_amountNeeded > _wantBal) {
+            _liquidatedAmount = _wantBal;
+            _loss = _amountNeeded - _wantBal;
+        } else {
+            _liquidatedAmount = _amountNeeded;
+        }
+    }
+    function _quoteMinAmountsRemove(uint256 amountLp) internal view returns (uint256 minAmountA, uint256 minAmountB){
+        (minAmountA, minAmountB) = IVeloRouter(AERO_ROUTER).quoteRemoveLiquidity(USDbC, DAI, true, POOL_FACTORY, amountLp);
+        minAmountA = minAmountA * slippage / 10000;
+        minAmountB = minAmountB * slippage / 10000;
+    }
+    function liquidateAllPositions()
+        internal override
+        returns (uint256 _amountFreed)
+    {
+        _claimAndSellRewards();
+
+        uint256 stakedAmount = balanceOfStaked();
+        IVeloGauge(AERO_GAUGE).withdraw(stakedAmount);
+        (uint256 minAmountA, uint256 minAmountB) = _quoteMinAmountsRemove(stakedAmount);
+        IVeloRouter(AERO_ROUTER).removeLiquidity(
+            USDbC,
+            DAI,
+            true,
+            stakedAmount,
+            minAmountA,
+            minAmountB,
+            address(this),
+            block.timestamp
+        );
+        _swapDaiToWant(IERC20(DAI).balanceOf(address(this)));
+        _amountFreed = want.balanceOf(address(this));
+    }
+
+    function prepareMigration(address _newStrategy) internal override {
+        uint256 assets = liquidateAllPositions();
+        want.safeTransfer(_newStrategy, assets);
+    }
+
+    function balanceOfStaked() public view returns (uint256 amount) {
+        amount = IVeloGauge(AERO_GAUGE).balanceOf(address(this));
+    }
+
+    function rewardss() public view returns (uint256 amount) {
```

Comment:
Typo: `rewardss()` -> `rewards()`

Reply:
there is a pub field in BaseStrategy, so => getRewards()